### DISCUSSION TITLE
Is there a way to use cell-voltage-tracking with JK's "PB" type Inverter BMS?

### DISCUSSION THREAD

**Discussion by NorthTown2022** [mr-manuel/venus-os_dbus-serialbattery#127] [General]
Is there a way to use cell-voltage-tracking with JK's "PB" type Inverter BMS?

I am waiting on two Yixiang DIY Battery Boxes and will soon switch over to the newer inverter type BMS from the older B2A20S20P (which has UART RS485 port). I am not sure how @mr-manuel's [cell-voltage-tracking](https://github.com/mr-manuel/venus-os_helpful-scripts/tree/master/cell-voltage-tracking)  can work without this UART port on the new BMS, at least without some modification. 

![image](https://github.com/user-attachments/assets/7c9f3bc6-d1f8-456a-bbc7-b006497b530a)

Can my existing JK RS485 adapter be modified, and used with one of the RJ45 RS485 ports on the JKPB2A16S20P breakout PCB or perhaps Bluetooth?
![20241214_135705](https://github.com/user-attachments/assets/c7929c3f-a027-44da-9659-dcb7dc070b97)

Has anyone made this work?

Are there any other options for cell tracking on a "PB" if this is not an option?

**Comment by mr-manuel** [comment]
I can't follow you. Just connect the RS485 of the JKBMS PB Model to a USB to RS485 adapter and you see all battery details.

See also https://mr-manuel.github.io/venus-os_dbus-serialbattery_docs/general/connect#jkbms-pb-model-also-know-as-jk-inverter-bms

**Comment by NorthTown2022** [comment]
Perhaps I was overthinking LOL. I will have PB model hooked up soon and test it out!
